src/components/TodoHeader.tsx

```
@@ -0,0 +1,100 @@
+import { useEffect, useRef, useState } from 'react';
+import { Errors, Todo } from '../types/Todo';
+import classNames from 'classnames';
+
+type Props = {
+  query: string;
+  todos: Todo[];
+  changeQuery: (query: string) => void;
+  setTodoError: (errorMessege: string) => void;
+  setSearchInput: (searchInput: React.RefObject<HTMLInputElement>) => void;
+  addNewTodo: () => void;
+  updateCompletedAllToDo: () => Promise<void>;
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  query,
+  todos,
+  changeQuery,
+  setTodoError,
+  addNewTodo,
+  setSearchInput,
+  updateCompletedAllToDo,
+}) => {
+  const [isShowToggleButton, setIShowToggleButton] = useState<boolean>(false);
+  const [isActiveTodos, setIsActiveTodos] = useState<boolean>(false);
+  const searchInput = useRef<HTMLInputElement>(null);
+  const setDisabledSearchInput = (isDisabled: boolean): void => {
+    if (searchInput.current) {
+      searchInput.current.disabled = isDisabled;
+    }
+  };
+
+  useEffect(() => {
+    setSearchInput(searchInput);
+    searchInput.current?.focus();
+  }, [setSearchInput]);
+
+  useEffect(() => {
+    if (todos.length !== 0) {
+      setIShowToggleButton(true);
+      setIsActiveTodos(!todos.some(tod => !tod.completed));
+    } else {
+      setIShowToggleButton(false);
+    }
+  }, [todos]);
+
+  const submitHandle = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (query.trim() === '') {
+      setTodoError(Errors.EMPTY);
+
+      return;
+    }
+
+    setDisabledSearchInput(true);
+    try {
+      await addNewTodo();
+      changeQuery('');
+    } catch (err) {}
+
+    setDisabledSearchInput(false);
+    searchInput.current?.focus();
+  };
+
+  const ToggleAllHandler = async () => {
+    try {
+      await updateCompletedAllToDo();
+    } catch (err) {}
```

why we need try {} catch () {} with empty catch block?